### PR TITLE
refactored procedure of selecting assets and events for bundling;

### PR DESCRIPTION
**Pull request by spherefoundry**

### CODE REVIEW

File: src/services/entity_repository.js [hunk 1]
```diff
@@ -115,37 +69,61 @@ export default class EntityRepository {
...
+      if (usedSize + size > bundleSizeInBytesLimit) {
+        return false;
+      }
+      usedSize += size;
+      return true;
+    };
 
-    const discardedEntities = this.discardEntitiesForBundling(orderedEntities, bundleItemsCountLimit, bundleSizeInBytesLimit);
-    await this.unsetEntitiesBundlesStubs(discardedEntities);
+    while (!(nextAsset === null && nextEvent === null) && (selectedAssets.length + selectedEvents.length < bundleItemsCountLimit)) {
```
Comment (Anze1m) [inline src/services/entity_repository.js]: 🍝😱🍝😱🍝😱

File: src/services/entity_repository.js [hunk 2]
```diff
@@ -46,67 +46,21 @@ export default class EntityRepository {
...
+    }
 
-  async unsetEntitiesBundlesStubs(entities) {
-    const update = {
-      $unset: {
-        'repository.bundleStubId': ''
-      }
-    };
-    await this.updateEntities(entities, update);
+    return asset;
```
[the remark this answers is not in the document]
Comment (vanruch) [reply src/services/entity_repository.js]: 👍

File: src/services/entity_repository.js (hunk 1 above)
Comment (Anze1m) [reply src/services/entity_repository.js]:
```suggestion
    while ((nextAsset !== null || nextEvent !== null) && (selectedAssets.length + selectedEvents.length < bundleItemsCountLimit)) {
```
Imo clearer - as it now explicitly says "Next asset or next event is not null and sum of already selected assets and events is not bigger than `bundleItemsCountLimit `"
Btw it should be `<` or `<=`?

File: src/services/entity_repository.js [hunk 3]
```diff
@@ -115,37 +69,61 @@ export default class EntityRepository {
...
+          break;
+        }
+
+        await this.db.collection('assets').updateOne({assetId: nextAsset.assetId}, updateBundleStubId);
+        selectedAssets.push(nextAsset);
+        nextAsset = await assetsCursor.next();
+      } else {
+        if (!fitsIntoSizeLimit(nextEvent)) {
+          break;
+        }
```
Comment (vanruch) [inline src/services/entity_repository.js]: This can be extracted out of _if_

**Pull request merged**